avogadro/io/lammpsformat.cpp

```diff
@@ -213,9 +213,15 @@ bool LammpsTrajectoryFormat::read(std::istream& inStream, Core::Molecule& mol)
     return false;
   }
   mol.setCoordinate3d(mol.atomPositions3d(), 0);
-  mol.setUnitCell(new UnitCell(Vector3(x_max - x_min, 0, 0),
-                               Vector3(tilt_xy, y_max - y_min, 0),
-                               Vector3(tilt_xz, tilt_yz, z_max - z_min)));
+  auto* uc = new UnitCell(Vector3(x_max - x_min, 0, 0),
+                          Vector3(tilt_xy, y_max - y_min, 0),
+                          Vector3(tilt_xz, tilt_yz, z_max - z_min));
+  if (!uc->isRegular()) {
+    appendError(
+      "'ITEM: BOX BOUNDS' does not give linear-independent lattive vectors");
+    return false;
```

Comment:
Okay, but if we `return false` then `uc` is leaked because it never gets added to the molecule.

`delete uc;` before `return false;`